### PR TITLE
GH-78707: Drop deprecated `pathlib.PurePath.[is_]relative_to()` arguments

### PR DESCRIPTION
**Pull request by barneygale**
Remove support for supplying additional positional arguments to `PurePath.relative_to()` and `is_relative_to()`. This has been deprecated since Python 3.12.

* Issue: gh-78707

----
📚 Documentation preview 📚: https://cpython-previews--118780.org.readthedocs.build/

### CODE REVIEW

File: Doc/whatsnew/3.14.rst [hunk 1]
```diff
@@ -106,6 +106,15 @@ Removed
   by Jelle Zijlstra in :gh:`118767`.)
 
 
+pathlib
+-------
```
Comment (hugovk) [inline Doc/whatsnew/3.14.rst:118]: Please can you move this above the "Others" section I added in https://github.com/python/cpython/pull/118799? We can list the modules alphabetically like in 3.13.

File: Misc/NEWS.d/next/Library/2024-05-08-18-59-19.gh-issue-78707._Lz1sw.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Drop support for passing additional positional arguments to
+:meth:`pathlib.PurePath.relative_to` and
+:meth:`~pathlib.PurePath.is_relative_to`.
```
Comment (hugovk) [inline Misc/NEWS.d/next/Library/2024-05-08-18-59-19.gh-issue-78707._Lz1sw.rst:3]: Tiny nit: What's New has `is_relative_to` before `relative_to`, let's have the same order in both.

**Review by hugovk — APPROVED** [review]
Thanks!

**Comment by barneygale** [comment]
Thank you!

**Pull request merged**